### PR TITLE
update usage doc for gatekeeper and starlark

### PR DESCRIPTION
**Pull request by mengqiy**
Also split constraint template and constraint from the resources to be validated.

### CODE REVIEW

**Comment by mengqiy** [comment]
Rebased on top of the Gatekeeper fn renaming PR.

The last commit contains the new changes since last review.

File: tests/gatekeeper/valid/policy.yaml [hunk 1]
```diff
@@ -32,7 +32,6 @@ kind: K8sBannedConfigMapKeysV1
 metadata:
   name: no-secrets-in-configmap
 spec:
-  enforcementAction: dryrun
```
Comment (frankfarzan) [inline tests/gatekeeper/valid/policy.yaml]: why is htis removed here and added above?

File: tests/gatekeeper/valid/to-validate.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
```
Comment (frankfarzan) [inline tests/gatekeeper/valid/to-validate.yaml:1]: why not just name this config-map.md?

File: tests/gatekeeper/valid/policy.yaml (hunk 1 above)
Comment (mengqiy) [reply tests/gatekeeper/valid/policy.yaml]: This is not true. GitHub diff appears to be not smart enough.
What happened is:
- tests/gatekeeper/valid/resources.yaml => tests/gatekeeper/valid/policy.yaml
- tests/gatekeeper/dryrun/resources.yaml => tests/gatekeeper/dryrun/policy.yaml

But GH diff thought it's:
- tests/gatekeeper/valid/resources.yaml => tests/gatekeeper/dryrun/policy.yaml
- tests/gatekeeper/dryrun/resources.yaml => tests/gatekeeper/valid/policy.yaml

**Review by frankfarzan — APPROVED** [review]

File: functions/go/starlark/README.md [hunk 3]
```diff
@@ -4,15 +4,20 @@
...
 
 <!--mdtogo-->
 
-### Synopsis
+### FunctionConfig
 
 <!--mdtogo:Long-->
 
-You can specify your Starlark script inline under field source like this:
+[Starlark] is a dialect of Python. It is commonly used as a configuration language. It is an untyped dynamic language with
```
Comment (frankfarzan) [inline functions/go/starlark/README.md:15]: This is part of the overview, not Function Config.

In general, this usage doc is very spartan and needs to be expanded.

**Pull request merged**